include/iotc_types.h

```diff
@@ -158,18 +146,15 @@ typedef void(iotc_user_subscription_callback_t)(
 
 /**
  * @enum iotc_crypto_key_union_type_t
- * @brief used to denote how the iotc_crypto_key_union_t union structure
- * should be evaluated.
- *
- *      - IOTC_CRYPTO_KEY_UNION_TYPE_PEM - if the key_pem.key pointer contains
- * a null terminated PEM key string.
- *      - IOTC_CRYPTO_KEY_UNION_TYPE_SLOT_ID - if the implementation is using
- * a secure element for private_key storage which indexes keys by slots,  then
- * this parameter will signal to the BSP which key storage slot to use for
- * cryptographic operations.
- *      - IOTC_CRYPTO_KEY_UNION_TYPE_CUSTOM - the uinion contains untyped data
- * which will be passed to the BSP for cryptographic operations. The BSP must
- * know the format of this data for its proper use.
+ * @brief Describe the iotc_crypto_key_union_t union structure.
+ *
+ *     - IOTC_CRYPTO_KEY_UNION_TYPE_PEM: the key_pem.key pointer contains
+ * a null-terminated PEM key string.
+ *     - IOTC_CRYPTO_KEY_UNION_TYPE_SLOT_ID: a secure element for private_key
```

Comment:
remove the word "private" here. we may reuse this definition for public keys, too.